bin/compile

```diff
@@ -211,6 +211,17 @@ cache_build() {
 header "Caching build"
```

Comment:
This changes a lot of the fixture files for tests which can be mostly ignored. The important changes are found in:

`bin/compile`
`lib/dependencies.sh`
`lib/environment.sh`
`lib/failure.sh`

With tests changes in:
`test/run`

which sadly gets shuffled to the bottom in this github view :(